### PR TITLE
✨ (rc): Add DeepSleep state

### PR DESCRIPTION
**Pull request by YannLocatelli**
## Requirements

- [x] #1238

### CODE REVIEW

**Review by HPezz — APPROVED** [review]
Nice work 👍

File: libs/RobotKit/include/RobotController.h [hunk 1]
```diff
@@ -294,6 +305,11 @@ class RobotController : public interface::RobotController
 		stopActuators();
 	}
 
+	void turnOffHardware() final
```
Comment (ladislas) [inline libs/RobotKit/include/RobotController.h:308]:
```suggestion
	void suspendHardwareForDeepSleep() final
	{
		log_info("Entering deep sleep");
	}
```

**Review by ladislas — APPROVED** [review]
LGTM 👍 
small naming suggestion as well

File: libs/RobotKit/include/RobotController.h (hunk 1 above)
Comment (YannLocatelli) [reply libs/RobotKit/include/RobotController.h]: Note. logs is not necessary since SM already display it

Comment (ladislas) [reply libs/RobotKit/include/RobotController.h]: yes I know, but SM displays the transitions not the "actual work".

the idea is just to have something print that we'll replace with the real implementation later to make sure the function is called when running.

```suggestion
	void suspendHardwareForDeepSleep() final
	{
		log_info("TO IMPLEMENT - configuring hardware for deepsleep");
	}
```

File: libs/RobotKit/include/RobotController.h [hunk 2]
```diff
@@ -292,6 +305,8 @@ class RobotController : public interface::RobotController
 		stopActuators();
 	}
 
+	void suspendHardwareForDeepSleep() final { log_info("Entering deep sleep"); }
```
Comment (ladislas) [inline libs/RobotKit/include/RobotController.h:308]: to answer your previous comment and to make it more obvious

```suggestion
	void suspendHardwareForDeepSleep() final { 	log_info("TO IMPLEMENT - configuring hardware for deep sleep"); }

```

**Review by ladislas — APPROVED** [review]
LGTM 👍 - great work! 💪 🙏

**Pull request merged**